mos-platform/cx16/cx16.h

```diff
@@ -433,10 +475,31 @@ void cx16_k_graph_set_font(void *fontaddr) __attribute__((leaf));
 void cx16_k_graph_set_window(unsigned int x, unsigned int y, unsigned int width, unsigned int height) __attribute__((leaf));
 int cx16_k_i2c_read_byte(unsigned char device, unsigned char offset) __attribute__((leaf)); // returns negative on error
 int cx16_k_i2c_write_byte(unsigned char device, unsigned char offset, unsigned char byte) __attribute__((leaf)); // return negative on error
-long cx16_k_joystick_get(unsigned char sticknum) __attribute__((leaf)); // returns $YYYYXXAA (see docs, result negative if joystick not present)
+
+/**
+ * Get joystick state using KERNAL routine `JOYSTICK_GET`.
+ *
+ * More information: https://github.com/X16Community/x16-docs
+ *
+ * @param joystick_num Keyboard joystick (0) or SNES controllers (1-4).
+ * @returns Struct with current status.
+ */
+inline struct JoyState cx16_k_joystick_get(const unsigned char joystick_num) {
+  struct JoyState s;
+  __attribute__((leaf)) asm volatile(
+      "JOYSTICK_GET = $FF56\n"
```

Comment:
There seems to be a disconnect between the symbol names here; only one has `__`.

Reply:
Good catch! `__JOYSTICK_GET` is defined one level up in `commodore/cbm_kernal.inc`. Removed local define.